src/traccuracy/metrics/_track_overlap.py

```diff
@@ -0,0 +1,119 @@
+"""This submodule implements routines for Track Purity (TP) and Target Effectiveness (TE) scores.
+
+Definitions (Bise et al., 2011; Chen, 2021; Fukai et al., 2022):
+
+- TE for a single ground truth track T^g_j is calculated by finding the predicted track T^p_k
+  that overlaps with T^g_j in the largest number of the frames and then dividing
+  the overlap frame counts by the total frame counts for T^g_j.
+  The TE for the total dataset is calculated as the mean of TEs for all ground truth tracks,
+  weighted by the length of the tracks.
+
+- TP is defined analogously, with T^g_j and T^p_j being swapped in the definition.
+"""
+
+from itertools import product
+from typing import TYPE_CHECKING, Any, List, Tuple
+
+from traccuracy._tracking_graph import TrackingGraph
+
+from ._base import Metric
+
+if TYPE_CHECKING:
+    from ._base import Matched
+
+
+class TrackOverlapMetrics(Metric):
+    """Calculate metrics for longest track overlaps.
+
+    - Target Effectiveness: fraction of longest overlapping prediction
+                            tracklets on each GT tracklet
+    - Track Purity : fraction of longest overlapping GT
+                     tracklets on each prediction tracklet
+
+    Args:
+        matched_data (Matched): Matched object for set of GT and Pred data
+        include_division_edges (bool, optional): If True, include edges at division.
+
+    """
+
+    supports_many_to_one = True
+
+    def __init__(self, matched_data: "Matched", include_division_edges: bool = True):
+        self.include_division_edges = include_division_edges
+        super().__init__(matched_data)
+
+    def compute(self):
+        gt_tracklets = self.data.gt_graph.get_tracklets(
+            include_division_edges=self.include_division_edges
+        )
+        pred_tracklets = self.data.pred_graph.get_tracklets(
+            include_division_edges=self.include_division_edges
+        )
+
+        gt_pred_mapping = self.data.mapping
```

Comment:
If you make these dictionaries, for example `gt_pred_mapping =  {gt_id: [pred_id, ...]}` and `pred_gt_mapping = {pred_id: [gt_id]}`, you can replace the map_node helper function in the _calc_overlap_score function with a simple dictionary lookup, and increase efficiency by not looping over all pairs to find matched edges.

Reply:
This is also done! I hope the implementation has become more efficient with this.